syntax = "proto3";

option go_package = "game/generated/pb/game";

import "proto/common/empty.proto";
import "proto/common/common.proto";

option cc_generic_services = true;

message RegisterSceneRequest
{
	uint32 scene_node_id = 1;
	repeated RoomInfoPBComponent scenes_info = 2;
}

message RegisterSceneResponse
{
}

message UnRegisterSceneRequest
{
	uint32 scene_node_id = 1;
	uint32 scene = 2;
	uint32 reason = 3;
}

//正常或者顶号进入场景
//断线重连进入场景,断线重连分时间
service CentreScene
{
  rpc RegisterScene (RegisterSceneRequest) returns (RegisterSceneResponse);
  rpc UnRegisterScene (UnRegisterSceneRequest) returns (Empty);
}